### PR TITLE
Improves docs

### LINKED ISSUE #4: Improve Haddocks
I have a PR with various haddock improvements but it would be amazing if someone more knowledgeable then myself could help on this task. Examples are especially valuable!

### PR DESCRIPTION
**Pull request by solomon-b**
This lib really could use some haddocks with clear examples. I've started that process but could use a lot of help and/or feedback to get it in good shape.

resolves #4

### CODE REVIEW

File: src/Control/Category/Tensor.hs [hunk 1]
```diff
@@ -10,29 +10,20 @@ import Data.Profunctor
...
-categories of two different monoidal structure, and has a pair of
-operations `t2 (f a) (f b) -> f (t1 a b)` and `i2 -> f i1`
-
--}
-
-class (Category cat1, Category cat2) => GBifunctor cat1 cat2 r t | t r -> cat1 cat2 where
-  gbimap :: a `cat1` b -> c `cat2` d -> t a c `r` t b d
+-- | A Bifunctor 't' is a binary type operator which allows you to map
+-- over both variables. GBifunctor is the same as the ordinary
+-- 'Data.Bifunctor.Bifunctor' class but we replace the '(->)'s with three different
```
Comment (masaeedu) [inline src/Control/Category/Tensor.hs:31]: Probably worth fleshing out here that we're talking about "arrows" or something. Maybe also some examples for arrows besides `->`.

File: src/Control/Category/Tensor.hs [hunk 2]
```diff
@@ -58,26 +49,36 @@ instance GBifunctor (Star Maybe) (Star Maybe) (Star Maybe) These where
...
-glmap :: GBifunctor cat1 cat2 r t => a `cat1` b -> t a c `r` t b c
+glmap :: GBifunctor cat1 cat2 cat3 t => cat1 a b -> cat3 (a `t` c) (b `t` c)
 glmap = flip gbimap id
 
-data Iso cat a b = Iso { fwd :: a `cat` b, bwd :: b `cat` a }
-
+data Iso cat a b = Iso { fwd :: cat a b, bwd :: cat b a }
+
+-- | An Associative Bifunctor is one whose type operator 't' is
+-- associative.
```
Comment (masaeedu) [inline src/Control/Category/Tensor.hs]: Somewhere in this haddock the word "isomorphism" should show up.

File: src/Control/Category/Tensor.hs [hunk 3]
```diff
@@ -104,6 +105,22 @@ instance Associative These (->) where
     , bwd = these (grmap This) (That . That) (flip $ grmap . flip These)
     }
 
+-- | A Tensor is an Associative Bifunctor 't' equipped with an
+-- identity type 'i' such that the left and right unit laws hold.
```
Comment (masaeedu) [inline src/Control/Category/Tensor.hs]: Maybe

> equipped with an identity type `i` _and a pair of left and right unit isomorphisms_

File: src/Control/Category/Tensor.hs (hunk 2 above)
Comment (solomon-b) [reply src/Control/Category/Tensor.hs]: Good call. I added a haddock to `Iso`

File: src/Control/Category/Tensor.hs (hunk 1 above)
Comment (solomon-b) [reply src/Control/Category/Tensor.hs:31]: Do you have a good example for an arrow other then `->`?

Comment (masaeedu) [reply src/Control/Category/Tensor.hs:31]: Sure. One example is `<-`, i.e. `Op`.

Comment (solomon-b) [reply src/Control/Category/Tensor.hs:31]: lmao touche

**Comment by solomon-b** [comment]
I rebased this PR off of `main`. We could either merge it as is or try to improve it before merging. Thoughts?

**Comment by solomon-b** [comment]
I'm merging this now. It isn't a complete set of haddocks but it is still an improvement over the current situation.

**Pull request merged**